src/Reflection/ReflectionClass.php

```diff
@@ -1238,4 +1244,17 @@ public function removeProperty($propertyName)
 
         return false;
     }
+
+    private function findTypeFromAst($namespace, $locatedSource, $type)
```

Comment:
Or, inject the context into the `FindType*` classes constructor, and have those injected individually; I'd prefer to not have them all bunched up into one class; they although they serve the same logical purpose, their semantics and usage is still subtly different I think. I'm open to ideas though.